lib/components/recordingSession/recorder.dart

```diff
@@ -28,94 +31,134 @@ class Recorder extends StatefulWidget {
-      final String date = DateFormat('yyyy-MM-ddThh:mm').format(now);
-      final Directory tempDir = await getTemporaryDirectory();
-      setState(() {
-        directory = tempDir;
-      });
-      print('tempdir: $tempDir');
-      final String path = await flutterSound.startRecorder(
-          uri: tempDir.path +
-              '/' +
-              date.toString() +
-              '_' +
-              userId.toString() +
-              '_sound.aac',
-          codec: t_CODEC.CODEC_AAC);
-      print('path: $path');
-      _recorderSubscription = flutterSound.onRecorderStateChanged.listen(
-        (RecordStatus e) {
-          final DateTime date = DateTime.fromMillisecondsSinceEpoch(
-              e.currentPosition.toInt(),
-              isUtc: true);
-          final String txt = DateFormat('mm:ss:SS', 'en_US').format(date);
-          setState(
-            () {
-              _recorderTxt = txt.substring(0, 8);
-            },
-          );
-        },
+      await startAudioSession();
+      final PermissionStatus status = await Permission.microphone.request();
+      if (status != PermissionStatus.granted) {
+        throw RecordingPermissionException('Microphone permission not granted');
+      }
+
+      final String path =
+          '${tempDir.path}/${now}_user_${userId}_recording${ext[_codec.index]}';
+
+      await recorderModule.startRecorder(
+        toFile: path,
+        codec: _codec,
+        bitRate: 16000,
+        sampleRate: 16000,
+        audioSource: AudioSource.voice_communication,
       );
-      _dbPeakSubscription = flutterSound.onRecorderDbPeakChanged.listen(
-        (double value) {
-          print('got update -> $value');
-          setState(
-            () {
-              _dbLevel = value;
-            },
-          );
+      print('startRecorder');
+      print('Path: ' + path);
+
+      _recorderSubscription = recorderModule.onProgress.listen(
+        (RecordingDisposition e) {
+          if (e != null && e.duration != null) {
+            final DateTime date = DateTime.fromMillisecondsSinceEpoch(
+                e.duration.inMilliseconds,
+                isUtc: true);
+            final String formattedDate =
+                DateFormat('mm:ss:SS', 'en_US').format(date);
+
+            //print('got update -> ${e.decibels}');
+
+            setState(() {
+              _recorderTxt = formattedDate.substring(0, 8);
+              _dbLevel = e.decibels;
+            });
+          }
         },
       );
 
-      setState(
-        () {
-          _isRecording = true;
-        },
-      );
+      setState(() {
+        savePath = path;
+        _isRecording = true;
+      });
     } catch (err) {
       print('startRecorder error: $err');
+      setState(() {
+        stopRecorder();
+        _isRecording = false;
+        if (_recorderSubscription != null) {
+          _recorderSubscription.cancel();
+          _recorderSubscription = null;
+        }
```

Comment:
use cancelAudioSubscription here rather than this

Reply:
lol wait did you comment on your own PR? @tonynguyen98

Reply:
maybe, haha. I saw this mistake earlier and just wanted to include it in with the rest of the fixes